src/multi_vector_simulator/F1_plotting.py

```diff
@@ -855,52 +856,70 @@ def create_plotly_flow_fig(
         figure object
     """
 
-    fig = go.Figure()
-    styling_dict = get_fig_style_dict()
-    styling_dict["gridwidth"] = 1.0
+    def create_plot(
+        flow_list, filename=file_name, y_legend=y_legend, plot_title=plot_title
+    ):
+        fig = go.Figure()
+        styling_dict = get_fig_style_dict()
+        styling_dict["gridwidth"] = 1.0
+
+        for i, asset in enumerate(flow_list):
+            fig.add_trace(
+                go.Scatter(
+                    x=df_plots_data["timestamp"],
+                    y=df_plots_data[asset],
+                    mode="lines",
+                    line=dict(color=get_color(i, color_list), width=2.5),
+                    name=asset,
+                )
+            )
 
-    assets_list = list(df_plots_data.columns)
-    assets_list.remove("timestamp")
+        fig.update_layout(
+            xaxis_title=x_legend,
+            yaxis_title=y_legend,
+            font_family="sans-serif",
+            template="simple_white",
+            xaxis=styling_dict,
+            yaxis=styling_dict,
+            title={
+                "text": plot_title,
+                "y": 0.90,
+                "x": 0.5,
+                "font_size": 23,
+                "xanchor": "center",
+                "yanchor": "top",
+            },
+            legend=dict(y=0.5, traceorder="normal", font=dict(color="black"),),
+        )
 
-    for i, asset in enumerate(assets_list):
-        fig.add_trace(
-            go.Scatter(
-                x=df_plots_data["timestamp"],
-                y=df_plots_data[asset],
-                mode="lines",
-                line=dict(color=get_color(i, color_list), width=2.5),
-                name=asset,
+        if file_path is not None:
+            # Function call to save the Plotly plot to the disk
+            save_plots_to_disk(
+                fig_obj=fig,
+                file_path=file_path,
+                file_name=filename,
+                width=1200,
+                height=600,
+                scale=5,
             )
-        )
+        return fig
```

Comment:
```suggestion
        return fig
```
Doesnt this exit the function, eventhough you have more code you want to use below?

Reply:
No, it only exits from the inner function `create_plot()`.